src/server/dflycmd.cc

```diff
@@ -390,10 +390,23 @@ void DflyCmd::ClusterConfig(CmdArgList args, ConnectionContext* cntx) {
     return rb->SendError("Invalid JSON cluster config", kSyntaxErrType);
   }
 
-  if (!sf_->cluster_config()->SetConfig(json.value())) {
+  auto deleted_slot_ids = sf_->cluster_config()->SetConfig(json.value());
+  if (!deleted_slot_ids) {
     return rb->SendError("Invalid cluster configuration.");
   }
 
+  // Delete old slots data.
+  if (!deleted_slot_ids.value().empty()) {
```

Comment:
Nit: you could use `deleted_slot_ids->empty()` here (i.e. drop the `.value().` part for a `->`)